src/main/java/frc/robot/subsystems/HoodSubsystem.java

```diff
@@ -0,0 +1,160 @@
+    m_encoder = m_motor.getEncoder();
+
+    // PID coefficients
+    kP = Constants.HOOD_kP;
+    kI = Constants.HOOD_kI;
+    kD = Constants.HOOD_kD;
+    kIz = Constants.HOOD_kIz;
+    kFF = Constants.HOOD_kFF;
+    kMaxOutput = Constants.HOOD_kMaxOutput;
+    kMinOutput = Constants.HOOD_kMinOutput;
+
+    // set PID coefficients
+    m_pidController.setP(kP);
+    m_pidController.setI(kI);
+    m_pidController.setD(kD);
+    m_pidController.setIZone(kIz);
+    m_pidController.setFF(kFF);
+    m_pidController.setOutputRange(kMinOutput, kMaxOutput);
+
+    // display PID coefficients on SmartDashboard
+    SmartDashboard.putNumber("P Gain", kP);
+    SmartDashboard.putNumber("I Gain", kI);
+    SmartDashboard.putNumber("D Gain", kD);
+    SmartDashboard.putNumber("I Zone", kIz);
+    SmartDashboard.putNumber("Feed Forward", kFF);
+    SmartDashboard.putNumber("Max Output", kMaxOutput);
+    SmartDashboard.putNumber("Min Output", kMinOutput);
+    SmartDashboard.putNumber("Set Rotations", rotations); 
+
+  }
+
+  @Override
+  public void periodic() {
+    // This method will be called once per scheduler run
+    // read PID coefficients from SmartDashboard
+    double p = SmartDashboard.getNumber("P Gain", 0);
+    double i = SmartDashboard.getNumber("I Gain", 0);
+    double d = SmartDashboard.getNumber("D Gain", 0);
+    double iz = SmartDashboard.getNumber("I Zone", 0);
+    double ff = SmartDashboard.getNumber("Feed Forward", 0);
+    double max = SmartDashboard.getNumber("Max Output", 0);
+    double min = SmartDashboard.getNumber("Min Output", 0);
+    // double rotations = SmartDashboard.getNumber("Set Rotations", 0);
+
+    // if PID coefficients on SmartDashboard have changed, write new values to
+    // controller
+    if ((p != kP)) {
+      m_pidController.setP(p);
+      kP = p;
+    }
+    if ((i != kI)) {
+      m_pidController.setI(i);
+      kI = i;
+    }
+    if ((d != kD)) {
+      m_pidController.setD(d);
+      kD = d;
+    }
+    if ((iz != kIz)) {
+      m_pidController.setIZone(iz);
+      kIz = iz;
+    }
+    if ((ff != kFF)) {
+      m_pidController.setFF(ff);
+      kFF = ff;
+    }
+    if ((max != kMaxOutput) || (min != kMinOutput)) {
+      m_pidController.setOutputRange(min, max);
+      kMinOutput = min;
+      kMaxOutput = max;
+    } 
+
+    if (Controller.getAButtonPressed()) {
+      rotations = 1.2;
+    }
+    if (Controller.getBButtonPressed()) {
+      rotations = 1;
+    }
+    if (Controller.getYButtonPressed()) {
+      rotations = 0.2;
+    }
+    if (Controller.getXButtonPressed()) {
+      rotations = 0;
+    }
+
+  
+    double position = m_encoder.getPosition();
+    
+
+    if (position < Constants.HOOD_min) {
+      position = Constants.HOOD_min;
+    }
+    if (position > Constants.HOOD_max) {
+      position = Constants.HOOD_max;
+    }
+
+    
+    //pid does this, dont need
+    if (position <= rotations + 0.03 && position >= rotations - 0.03) {
```

Comment:
Comment this out. As your comment covers, PID should get the hood to the position we need if tuned correctly.